v22.1/crdb-internal.md

```diff
@@ -264,6 +265,404 @@ SELECT * FROM crdb_internal.cluster_contention_events;
+
+{{site.data.alerts.callout_success}}
+You can see the types and default values of columns in this and other tables using [`SHOW COLUMNS FROM {table}`](show-columns.html).
+{{site.data.alerts.end}}
+
+#### Cluster locks - basic example
+
+In this example, we'll use the [`SELECT FOR UPDATE`](select-for-update.html) statement to order two transactions by controlling concurrent access to a table. Then, we will look at the data in `cluster_locks` to see the locks being held by these transactions on the objects they are accessing.
+
+{% include {{page.version.version}}/sql/select-for-update-example-partial.md %}
+
+Now that we have two transactions both trying to update the `kv` table, let's query the data in `crdb_internal.cluster_locks`. We should see two locks:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT database_name, table_name, txn_id, ts, lock_key_pretty, lock_strength, granted, contended FROM crdb_internal.cluster_locks WHERE table_name = 'kv';
+~~~
+
+~~~
+  database_name | table_name |                txn_id                |             ts             | lock_key_pretty  | lock_strength | granted | contended
+----------------+------------+--------------------------------------+----------------------------+------------------+---------------+---------+------------
+  defaultdb     | kv         | d11a08c4-a3a2-4bdb-bf10-8d2373426faf | 2022-07-27 18:57:06.808046 | /Table/107/1/1/0 | Exclusive     |  true   |   true
+  defaultdb     | kv         | 34ebadb6-99f1-4547-b487-4b322506b7fe | 2022-07-27 18:57:13.173556 | /Table/107/1/1/0 | Exclusive     |  false  |   true
+(2 rows)
+~~~
+
+As expected, there are two locks. This is the case because:
+
+- The transaction with the [`SELECT FOR UPDATE`](select-for-update.html) query in Terminal 1 asked for an `Exclusive` lock on a row in the `defaultdb.kv` table, as shown in the `lock_strength` column. We can see that it was able to get that lock, since the `granted` column is `true`.
+- The transaction in Terminal 2 is also trying to lock the same row in the `kv` table with a `lock_strength` of `Exclusive`. However, the value of the `granted` column is `false`, which means it could not get the exclusive lock yet, and is waiting on the lock from the query in Terminal 1 to be released before it can proceed.
+
+Further, both transactions show the `contended` column as `true`, since these transactions are both trying to update rows in the `defaultdb.kv` table at the same time.
+
+The following more advanced query shows more information about lockholders, sessions, and waiting queries. This may be useful on a busy cluster for figuring out which transactions from which clients are trying to grab locks. Note that joining with `cluster_queries` will only show queries currently in progress.
```

Comment:
```suggestion
The following more complex query shows additional information about lockholders, sessions, and waiting queries. This may be useful on a busy cluster for figuring out which transactions from which clients are trying to grab locks. Note that joining with `cluster_queries` will only show queries currently in progress.
```
Add "lockholder" and "lockholders" to `netlify/vale/vocab.txt` to get rid of the Vale error.

Reply:
Did it! thanks